File: Programmers/phone_book.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for(int i=0;i<phone_book.length-1; i++){
            for(int j=i+1; j<phone_book.length; j++){
                if(phone_book[i].length() < phone_book[j].length()){
                    if(phone_book[i].equals(phone_book[j].substring(0,phone_book[i].length())))
                         return false;
                }      
            }
        }
        return answer;
    }
}

File: Code/Programmers/phone_book.py
# new code
def solution(phone_book):
    phone_book.sort() # 효율성 8,9번 통과의 핵심
    min = len(phone_book[0])
    prefix = []
    for i,p in enumerate(phone_book):
        if min >= len(p):
            prefix.append(phone_book.pop(i))
    for i in range(len(prefix)):
        for p in phone_book:
            if p.startswith(prefix[i]) :
                return False
    return True
    
# old code
def solution(phone_book):
    answer = False
    min_length = 0
    min_content = phone_book[0]
    for i in phone_book:
        if min_length > len(i):
            min_length = len(i)
            min_content = i
    for i in phone_book :
        if i[:min_length] == min_content :
            answer = False
        else:
            answer = True
    return answer